src/twisted/application/twist/_options.py

```diff
@@ -188,6 +187,8 @@ def subCommands(self):
 
 
     def postOptions(self):
+        self.installReactor()
+
```

Comment:
It looks like this is too late.  The unit test probably passes because it doesn't have the same condition as the real implementation: that if no reactor is installed, importing twisted.internet.reactor installs one and attempting to install a second one fails.  Instead, the unit test doesn't notice that the default reactor gets installed by an accidental import somewhere.

This is a good example of why it might be better to make the tests more realistic. :)

I had a hunch that `postOptions` might be too late so I ran `twist` manually and indeed:

```
$ twist --reactor poll web
Traceback (most recent call last):
  File "/home/exarkun/Environments/twisted-dev/bin/twist", line 11, in <module>
    load_entry_point('Twisted', 'console_scripts', 'twist')()
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/application/twist/_twist.py", line 132, in main
    options = cls.options(argv)
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/application/twist/_options.py", line 159, in parseOptions
    Options.parseOptions(self, options=options)
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/python/usage.py", line 277, in parseOptions
    self.postOptions()
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/application/twist/_options.py", line 190, in postOptions
    self.installReactor()
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/application/twist/_options.py", line 77, in installReactor
    self["reactor"] = installReactor(name)
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/application/reactors.py", line 82, in installReactor
    installer.install()
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/application/reactors.py", line 60, in install
    namedAny(self.moduleName).install()
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/internet/pollreactor.py", line 186, in install
    installReactor(p)
  File "/home/exarkun/Work/LeastAuthority/twisted/src/twisted/internet/main.py", line 32, in installReactor
    raise error.ReactorAlreadyInstalledError("reactor already installed")
twisted.internet.error.ReactorAlreadyInstalledError: reactor already installed
```

Perhaps immediately following the super-class call to `parseOptions` would be a better place.  Or, even earlier, inside `opt_reactor` itself (isn't that where twistd and trial do it?)